### PR TITLE
[MHV-44568] Updated Navigation after sending a message

### PR DESCRIPTION
**Pull request by vsaleem**
## Summary
MHV-44568: Updated Navigation after sending a message
- User will be redirected to the original folder from where the message was sent from.
- Originally the user was sent to the Inbox or Sent folder regardless of where the message was sent from.

MHV-44474: User Unable to Remove Custom Folder
- When a user removes an EMPTY folder they will be presented with a modal to delete the folder.
- When a user attempts to remove a folder with messages, they will be present with a modal that tells them to empty the folder before they can delete it.

## Related issue(s)
https://jira.devops.va.gov/browse/MHV-44568
https://jira.devops.va.gov/browse/MHV-44474

## What areas of the site does it impact?

*VA.GOV - Secure Messaging*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]

File: src/applications/mhv/secure-messaging/tests/containers/FolderThreadListView.unit.spec.jsx [hunk 1]
```diff
@@ -97,19 +106,47 @@ describe('Folder Thread List View container', () => {
...
+      `/folders/${customFolder.folderId}`,
+    );
+
+    await waitFor(() => {
+      expect(global.document.title).to.equal(
+        `${customFolder.name} ${PageTitles.PAGE_TITLE_TAG}`,
+      );
+      const folderName = screen.getByRole('heading', { level: 1 });
+      expect(folderName).to.exist;
+      expect(folderName).to.have.text(customFolder.name);
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/tests/containers/FolderThreadListView.unit.spec.jsx:145]: not critical, but 3 lines above can be optimized in 1 line like so
```suggestion
      expect(screen.getByText(customFolder.name, { selector: 'h1' })).to.exist;
```

File: src/applications/mhv/secure-messaging/tests/containers/FolderThreadListView.unit.spec.jsx [hunk 2]
```diff
@@ -97,19 +106,47 @@ describe('Folder Thread List View container', () => {
...
+      const folderDescription = screen.getByTestId('folder-description');
+      expect(folderDescription).to.exist;
+      expect(folderDescription).to.have.text(DefaultFolders.CUSTOM_FOLDER.desc);
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/tests/containers/FolderThreadListView.unit.spec.jsx:148]: same suggestion as above

File: src/applications/mhv/secure-messaging/tests/containers/FolderThreadListView.unit.spec.jsx [hunk 3]
```diff
@@ -97,19 +106,47 @@ describe('Folder Thread List View container', () => {
...
+    const screen = setup(initialStateTrash, Paths.DELETED);
 
     await waitFor(() => {
       expect(global.document.title).to.equal(
         `Trash ${PageTitles.PAGE_TITLE_TAG}`,
       );
     });
+    const folderName = screen.getByRole('heading', { level: 1 });
+    expect(folderName).to.exist;
+    expect(folderName).to.have.text(DefaultFolders.DELETED.header);
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/tests/containers/FolderThreadListView.unit.spec.jsx:118]: same suggestion as on lines 143-145

File: src/applications/mhv/secure-messaging/tests/containers/FolderThreadListView.unit.spec.jsx [hunk 4]
```diff
@@ -97,19 +106,47 @@ describe('Folder Thread List View container', () => {
...
+
+    const folderDescription = screen.getByTestId('folder-description');
+    expect(folderDescription).to.exist;
+    expect(folderDescription).to.have.text(DefaultFolders.DELETED.desc);
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/tests/containers/FolderThreadListView.unit.spec.jsx:122]: same suggestion as on lines 143-145

File: src/applications/mhv/secure-messaging/components/ManageFolderButtons.jsx [hunk 5]
```diff
@@ -65,7 +65,7 @@ const ManageFolderButtons = () => {
 
   const openDelModal = () => {
     dispatch(closeAlert());
-    if (messages?.length > 0) {
+    if (threads.threadList.length > 0) {
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/ManageFolderButtons.jsx:68]: we probably should cover this logic in a unit test as well. Please pardon if you already do, I may have missed

**Comment by oleksii-morgun-va** [comment]
Great additions! I think we should strive to cover the scenarios with 'Delete Folder' scenarios in unit tests too. Particularly since this issue has stroke us back this time

File: src/applications/mhv/secure-messaging/tests/containers/ThreadDetails.unit.spec.jsx [hunk 6]
```diff
@@ -185,6 +185,12 @@ describe('Thread Details container', () => {
 
     const state = {
       sm: {
+        folders: {
+          folder: {
+            folderId: 0,
+            name: 'Inbox',
+          },
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/tests/containers/ThreadDetails.unit.spec.jsx:192]: We should have these objects in mocks in `/fixtures` that we can re-use here and in other tests too

**Pull request closed without merging**